types/src/key.rs

```diff
@@ -561,6 +593,10 @@ impl FromBytes for Key {
                 let (addr, rem) = DictionaryAddr::from_bytes(remainder)?;
                 Ok((Key::Dictionary(addr), rem))
             }
+            tag if tag == KeyTag::SystemContractRegistry as u8 => {
+                let (_, rem): ([u8; 32], &[u8]) = FromBytes::from_bytes(remainder)?;
+                Ok((Key::SystemContractRegistry, rem))
+            }
             _ => Err(Error::Formatting),
         }
     }
```

Comment:
I can't apply the comment on the appropriate line, but `impl Distribution<Key> for Standard` should be updated to allow the new key variant to be randomly generated.